### PR TITLE
[WIP] New experimental P/Invoke checker.

### PR DESCRIPTION
**Pull request by baulig**

### CODE REVIEW

File: mcs/tools/pinvoke-checker/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
+thisdir = tools/pinvoke-checker
+SUBDIRS =
+include ../../build/rules.make
+
+PROGRAM = pinvoke-checker.exe
+NO_INSTALL = yes
+
+API = $(filter basic build, $(PROFILE))
```
Comment (marek-safar) [inline mcs/tools/pinvoke-checker/Makefile]: This should not be needed as we want to run it as part of the test phase

File: mcs/tools/pinvoke-checker/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ifdef MCS_MODE
+LIB_REFS += Mono.Cecil.Mdb
+endif
+
+include ../../build/executable.make
+
+IOS_BUILD_OUT = ../../../sdks/out/ios-sim64-release
+MAC_BUILD_OUT = ../../../../../builds/install/mac64
+
+MARTIN_MONOTOUCH_ARGS = \
```
Comment (marek-safar) [inline mcs/tools/pinvoke-checker/Makefile]: Hmm, I suspect this should be converted to SDKs test targets

File: mcs/tools/pinvoke-checker/pinvoke-checker.cs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+		Console.WriteLine ($"Added {symbols.Count} symbols from {library.Path}.");
+
+		return true;
+	}
+
+	void CheckAssembly ()
+	{
+		var readerParameters = new ReaderParameters {
+			ReadSymbols = true,
```
Comment (marek-safar) [inline mcs/tools/pinvoke-checker/pinvoke-checker.cs]: Why do we need debug symbols?

File: mcs/tools/pinvoke-checker/pinvoke-checker.cs [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+		var function = method.PInvokeInfo.EntryPoint;
+		switch (function) {
+			case "dlopen":
+			case "dlsym":
+			case "dlclose":
+				return;
+		}
+
+		if (Platform.IsMacOS) {
```
Comment (marek-safar) [inline mcs/tools/pinvoke-checker/pinvoke-checker.cs]: Use public API for this available in RuntimeInformation.IsOSPlatform

**Comment by steveisok** [comment]
@marek-safar I'm assuming since this has been around for a while that it's no longer needed?

**Comment by baulig** [comment]
@steveisok xamarin-macios has their own logic in their `mtouch` / `mmp` tools.

**Pull request closed without merging**